cmplxs/cmplxs.go

```diff
@@ -0,0 +1,679 @@
+		if f(val) {
+			inds = append(inds, i)
+			nFound++
+			if nFound == k {
+				return inds, nil
+			}
+		}
+	}
+	// Finished iterating over the loop, which means k elements were not found.
+	return inds, errors.New("cmplxs: insufficient elements found")
+}
+
+// HasNaN returns true if the slice s has any values that are NaN and false
+// otherwise.
+func HasNaN(s []complex128) bool {
+	for _, v := range s {
+		if cmplx.IsNaN(v) {
+			return true
+		}
+	}
+	return false
+}
+
+// LogSpan returns a set of n equally spaced points in log space between,
+// l and u where N is equal to len(dst). The first element of the
+// resulting dst will be l and the final element of dst will be u.
+// Panics if len(dst) < 2
+// Note that this call will return NaNs if either l or u are negative, and
+// will return all zeros if l or u is zero.
+// Also returns the mutated slice dst, so that it can be used in range, like:
+//
+//     for i, x := range LogSpan(dst, l, u) { ... }
+func LogSpan(dst []complex128, l, u complex128) []complex128 {
+	Span(dst, cmplx.Log(l), cmplx.Log(u))
+	for i := range dst {
+		dst[i] = cmplx.Exp(dst[i])
+	}
+	return dst
+}
+
+// MaxAbs returns the maximum absolute value in the input slice.
+// If the slice is empty, MaxAbs will panic.
+func MaxAbs(s []complex128) complex128 {
+	return s[MaxAbsIdx(s)]
+}
+
+// MaxAbsIdx returns the index of the maximum absolute value in the input slice. If several
+// entries have the maximum absolute value, the first such index is returned. If the slice
+// is empty, MaxAbsIdx will panic.
+func MaxAbsIdx(s []complex128) int {
+	if len(s) == 0 {
+		panic("cmplxs: zero slice length")
+	}
+	max := math.NaN()
+	var ind int
+	for i, v := range s {
+		if cmplx.IsNaN(v) {
+			continue
+		}
+		if a := cmplx.Abs(v); a > max || math.IsNaN(max) {
+			max = a
+			ind = i
+		}
+	}
+	return ind
+}
+
+// MinAbs returns the minimum absolute value in the input slice.
+// If the slice is empty, MinAbs will panic.
+func MinAbs(s []complex128) complex128 {
+	return s[MinAbsIdx(s)]
+}
+
+// MinAbsIdx returns the index of the minimum absolute value in the input slice. If several
+// entries have the minimum absolute value, the first such index is returned. If the slice
+// is empty, MinAbsIdx will panic.
+func MinAbsIdx(s []complex128) int {
+	if len(s) == 0 {
+		panic("cmplxs: zero slice length")
+	}
+	min := math.NaN()
+	var ind int
+	for i, v := range s {
+		if cmplx.IsNaN(v) {
+			continue
+		}
+		if a := cmplx.Abs(v); a < min || math.IsNaN(min) {
+			min = a
+			ind = i
+		}
+	}
+	return ind
+}
+
+// Mul performs element-wise multiplication between dst
+// and s and stores the value in dst. Panics if the
+// lengths of s and t are not equal.
+func Mul(dst, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	for i, val := range s {
+		dst[i] *= val
+	}
+}
+
+// MulTo performs element-wise multiplication between s
+// and t and stores the value in dst. Panics if the
```

Comment:
"values" or "result"

Reply:
Done

Will also need to be done in floats.